builtin/logical/database/backend.go

```diff
@@ -27,6 +29,29 @@ const (
 	minRootCredRollbackAge = 1 * time.Minute
 )
 
+// metrics collection
+var (
+	gaugeSync = sync.Mutex{}
+	gauges    = map[string]*int32{}
+	gaugeKey  = []string{"secrets", "database", "backend", "connections", "count"}
+)
+
+func createGauge(name string) {
+	gaugeSync.Lock()
+	defer gaugeSync.Unlock()
+	gauges[name] = new(int32)
+}
+
+func addToGauge(dbType, version string, uuid string, amount int32) {
+	labels := []metrics.Label{{"type", dbType}, {"version", version}, {"backend", uuid}}
```

Comment:
`"version"` may cause some confusion when plugin versioning is introduced in the near future. In order of preference, I would suggest either omitting it or giving it a more detailed name. Is using v4 vs v5 of the plugin interface especially valuable to have in the metrics? The other two labels seem much more worthwhile to me.